### DISCUSSION TITLE
How can I create close transitions to Dialog?

### DISCUSSION THREAD

**Discussion by Klynger** [radix-ui/primitives#1534] [Help] [answered]
I'm trying to create a close transition to the Dialog, but I can't find how it should be done. I noticed that even the demo example in the documentation doesn't have an animation when closing the dialog. I know this thing works because I tested your storybook.

![](https://user-images.githubusercontent.com/8517023/179504562-ea362746-e362-4fea-acbd-bef0dce22d1a.gif)

I edited the codesandbox from the documentation with the changes I thought would be this work. Could you tell me what is wrong?

https://codesandbox.io/s/recursing-liskov-py39ii?file=/App.js:1488-1601

**Comment by andy-hook** [comment]
@Klynger You need to provide an additional exit animation in your `data-state="closed"` block which is the inverse of your "show"

https://codesandbox.io/s/dialog-animate-out-qr401v?file=/App.js

**Comment by Klynger** [comment, marked as answer]
Hey, @andy-hook . Thank you for the answer, but your solution doesn't work if I wrap the content in a `DialogPrimitive.Portal`. Is that expected?

**Reply by andy-hook** [reply]
I'm not sure I follow, that example is already using `DialogPrimitive.Portal` inside the `Content` component and it appears to work fine.

It sounds like you might be using portal twice in your case, perhaps double check there.

**Reply by Klynger** [reply]
Ok, that was a problem. But I'm trying to add a wrapper in the children of the portal to add a `z-index`, but when I do this, the animation stops working.

This is what I'm trying to do. Is that correct?
<img width="636" alt="image" src="https://user-images.githubusercontent.com/8517023/179553207-fe06c72f-6ecf-4161-8e78-d8b6f91ef9e7.png">

**Reply by andy-hook** [reply]
`Overlay` returns a node and accepts `className` so you can apply your `z-index` to it directly.

**Reply by benoitgrelard** [reply]
Yeah you can apply `z-index` directly to overlay and content.
